### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by redwing555**
In the following PR we worked on :

- Removing` current_user` method in `ApplicationController`, devise will provide us with one.
  - Install and set up devise.
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed passwords should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- Modifying the views in devise for registration and login to match the wireframes from the Sneak Peek .

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 class User < ApplicationRecord
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable, :validatable, :confirmable
```
Comment (SajjadAhmad14) [inline app/models/user.rb:2]: The user is unable to log in after creating an account. If we remove `:confirmable` then the user can log in successfully.
Kindly see the screenshot below which asks to confirm email first but this feature is not yet implemented. Please make sure the user can log in with a valid email and password.
confirm

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Houssam, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed.  :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/user.rb (hunk 1 above)
Comment (Lordkaito) [reply app/models/user.rb:2]: Hi, @SajjadAhmad14. We had a requirement saying that we had to make sure that the email is verified, reference here:
Screenshot_2022-01-25_21-50-27

If this is not correct, then what are we supposed to do with the confirmation email?

Comment (SajjadAhmad14) [reply app/models/user.rb:2]: Hi, so it's a requirement to verify email before signing in but it's not working in the project. Please make sure that the user is able to sign in after signing up.

**Review by uwadonat — APPROVED** [review]
**Hi @Team,** 

Your project is complete!  Congratulations on your effort to make things done! 🎉 
good-job


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by elyor-doniyorov** [comment]
## You can submit the project it is approved ✔️

**Pull request merged**